### ISSUE TITLE
Dependencies 'go.opencensus.io/exemplar' broken

### ISSUE THREAD

**Issue by kevinkirkup** [terraform-providers/terraform-provider-azurerm#3638]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

~v1.28.0 - v1.30.1
Not sure when it started...

### Affected Resource(s)

N/A Occurs when loading provider SDK

### Terraform Configuration Files

N/A SDK

### Debug Output

```
package github.com/pureport/terraform-provider-pureport/pureport (test)
	imports github.com/terraform-providers/terraform-provider-azurerm/azurerm
	imports github.com/Azure/azure-sdk-for-go/profiles/2017-03-09/resources/mgmt/resources
	imports github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2016-02-01/resources
	imports github.com/Azure/go-autorest/autorest
	imports github.com/Azure/go-autorest/autorest/adal
	imports github.com/Azure/go-autorest/tracing
	imports contrib.go.opencensus.io/exporter/ocagent
	imports go.opencensus.io/exemplar: unknown import path "go.opencensus.io/exemplar": cannot find module providing package go.opencensus.io/exemplar
```

### Expected Behavior

`go get github.com/pureport/terraform-providers/terraform-provider-azurerm` should be able to download all dependencies.

### Actual Behavior

`go.opencensus.io/exemplar` isn't found.

### Steps to Reproduce

1. Create a project that imports the provider SDK
1. `go get github.com/pureport/terraform-providers/terraform-provider-azurerm`
1. Build it.

### References

* #0000

**Comment by kevinkirkup** [comment]
Looks like `github.com/Azure/go-autorest` isn't using `go mod` and so the dependency tree is broken.

**Comment by kevinkirkup** [comment]
Or could be that "go.opencensus.io/exemplar" has been deprecated.

**Comment by kevinkirkup** [comment]
Override with autorest v12 seems to work as expected.

**Comment by tamalsaha** [comment]
https://github.com/Azure/go-autorest/pull/431 seemed to work for me.

**Comment by tombuildsstuff** [comment]
This should be fixed via the upcoming v0.7.0 of hashicorp/go-azure-helpers

**Comment by tombuildsstuff** [comment]
This should have been fixed via #4166 - closing
